lib/Service/DocumentService.php

```diff
@@ -206,10 +206,12 @@ public function writeDocumentState(int $documentId, string $content): void {
 	}
 
 	/**
-	 * @throws DoesNotExistException
 	 * @throws InvalidArgumentException
+	 * @throws NotFoundException
+	 * @throws NotPermittedException
+	 * @throws DoesNotExistException
 	 */
-	public function addStep(Document $document, Session $session, array $steps, int $version): array {
+	public function addStep(Document $document, Session $session, array $steps, int $version, ?string $token): array {
```

Comment:
```suggestion
	public function addStep(Document $document, Session $session, array $steps, int $version, ?string $shareToken): array {
```

Maybe we call it like that to make it more clear that this is not the text session token

Reply:
Agreed and changed :relaxed: